### PR TITLE
Core/Scripts Boss Ahune Clean-up

### PR DESCRIPTION
**Pull request by sirikfoll**
**Changes proposed**:
- Remove use of flags UNIT_FLAG_NON_ATTACKABLE and   UNIT_FLAG_REMOVE_CLIENT_CONTROL
- Replace with correct one, UNIT_FLAG_UNK_31, imunity part now is fully handled by auras
- Fix a problem with bugged orientation while casting SPELL_SYNCH_HEALTH
- Use std::crono notations
- Remove unneeded code
- Fix loot issue
- Fix possible crash
- Fix Ice Spear target selection

**Target branch(es)**: 335/6x

**Tests performed**: Built and tested

### CODE REVIEW

**Comment by tkrokli** [comment]
Are you stupid? Go learn C++

File: src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp [hunk 1]
```diff
@@ -169,19 +168,14 @@ class boss_ahune : public CreatureScript
     {
         boss_ahuneAI(Creature* creature) : BossAI(creature, DATA_AHUNE)
         {
-            Initialize();
-        }
-
-        void Initialize()
-        {
-            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_REMOVE_CLIENT_CONTROL);
+            me->SetControlled(true, UNIT_STATE_ROOT);
```
Comment (MitchesD) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: are you sure about this? really it should be only in ctor? shouldn't the NPC be rooted even after wipe for example? or do I miss something?

File: src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp [hunk 2]
```diff
@@ -333,23 +320,20 @@ class npc_frozen_core : public CreatureScript
...
                 _events.Reset();
                 DoCast(me, SPELL_ICE_SPEAR_AURA);
-                me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_REMOVE_CLIENT_CONTROL);
+                me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_IMMUNE_TO_PC);
             }
         }
 
         void UpdateAI(uint32 diff) override
         {
-            if (!UpdateVictim())
```
Comment (MitchesD) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: explanation why did you remove this?

File: src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp (hunk 1 above)
Comment (sirikfoll) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: After a wipe he despawn, this way I get the exact effect that i saw in blizz sniff, `Movement Flags: Root (2048)` in the creation packet.

File: src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp (hunk 2 above)
Comment (sirikfoll) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: Frozen Core is completely passive, he never needs to acquire or updates target and such

Comment (joschiwald) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: behaviour of UpdateVictim depends on reactstate, if aggressive, search target, no target -> evade, if passive, evade on empty threatlist

Comment (sirikfoll) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: true, but I don't need him to evade either

File: src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp [hunk 3]
```diff
@@ -268,16 +262,16 @@ class boss_ahune : public CreatureScript
...
+            events.ScheduleEvent(EVENT_SYNCH_HEALTH, Seconds(3));
         }
 
         void Submerge()
         {
             if (Creature* frozenCore = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FROZEN_CORE)))
                 frozenCore->AI()->DoAction(ACTION_AHUNE_RETREAT);
             me->RemoveAurasDueToSpell(SPELL_AHUNES_SHIELD);
-            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE);
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNK_31);
```
Comment (Kittnz) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: What does UNIT_FLAG_UNK_31 do?

Comment (sirikfoll) [inline src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp]: No ideia, it's supposed to have it

**Comment by Aokromes** [comment]
This can be merged?
